drivers/storage/port/scsiport/scsi.c

```diff
@@ -683,8 +682,23 @@ SpiProcessCompletedRequest(
     /* Flush adapter if needed */
     if (SrbInfo->BaseOfMapRegister)
     {
-        /* TODO: Implement */
-        ASSERT(FALSE);
+        if (SrbInfo->ScatterGather)
```

Comment:
Nit: you could combine this with the above into `if (SrbInfo->BaseOfMapRegister && SrbInfo->ScatterGather)` now.

Reply:
I combined it, thanks for the suggestions :)